control_panel_api/tests/test_views.py

```diff
@@ -128,15 +125,24 @@ def test_detail(self):
         self.assertIn('arn', response.data)
         self.assertEqual(4, len(response.data))
 
-    def test_delete(self):
+    @patch('boto3.client')
+    def test_delete(self, mock_client):
         response = self.client.delete(
             reverse('s3bucket-detail', (self.fixture.id,)))
         self.assertEqual(HTTP_204_NO_CONTENT, response.status_code)
 
         response = self.client.get(reverse('s3bucket-detail', (self.fixture.id,)))
         self.assertEqual(HTTP_404_NOT_FOUND, response.status_code)
 
-    def test_create_when_valid_data(self):
+    @patch('control_panel_api.services.delete_bucket')
+    @patch('control_panel_api.services.delete_bucket_policies')
+    def test_delete_calls_apis(self, mock_delete_bucket_policies, mock_delete_bucket):
+        self.client.delete(reverse('s3bucket-detail', (self.fixture.id,)))
+        mock_delete_bucket_policies.assert_called()
+        mock_delete_bucket.assert_called()
```

Comment:
As I said, let's just delete the IAM policies and not the S3 bucket itself for the time being (we can change this later if really requested)

Reply:
Cool, just removed it. Another example of a good test - removing the delete call in the view caused the tests fail.